### ISSUE TITLE
CI artifact names conflict

### ISSUE THREAD

**Issue by FilipRazek** [sharkdp/hexyl#215]
In the current CI/CD setup, the artifacts generated by `arm-unknown-linux-gnueabihf` and `arm-unknown-linux-musleabihf` have the same name, `hexyl_0.14.0_armhf.deb`. The default behavior from `upload-artifact` seems to be to fail in this case (which was probably a [breaking change in v4](https://github.com/actions/upload-artifact/issues/478)).

However, the CI/CD currently depends on [upload-artifact@master](https://github.com/actions/upload-artifact/tree/master/), which is not the latest branch (main is).

I do not understand why the CI breaks now, but it would make sense to:
- Change the artifact names for different architectures
- Update the dependency to upload-artifact@main (or at least a more recent version)

**Comment by sharkdp** [comment]
Yes. Thank you. I did similar changes in my other projects (although I didn't notice the branch name change…)
